### PR TITLE
Resolve 'Sensor data type should be controlled by daq_common.h'

### PR DESCRIPTION
**Pull request by MellingJJ**
@CameronLR Could you review this?

### CODE REVIEW

File: Teensy/src/gpio/gpio.cpp [hunk 1]
```diff
@@ -11,59 +11,59 @@
...
 
-static int32_t gpio_getWheelSpeed();
-static uint32_t gpio_getEngineRevs();
-static int32_t gpio_getDamperPosition();
-static int32_t gpio_getGearPosition();
-static uint32_t gpio_getSteeringWheelPosition();
-static int32_t gpio_getStrain();
+static daq_WheelSpeed_t gpio_getWheelSpeed();
+static daq_EngineRev_t gpio_getEngineRevs();
+static daq_DamperPos_t gpio_getDamperPosition();
```
Comment (CameronLR) [inline Teensy/src/gpio/gpio.cpp:16]: Needs it's own datatype to store all 4 damper positions

File: Teensy/src/gpio/gpio.cpp [hunk 2]
```diff
@@ -11,59 +11,59 @@
 #include <Arduino.h>
 #include "gpio.h"
 
-static int32_t gpio_getWheelSpeed();
-static uint32_t gpio_getEngineRevs();
-static int32_t gpio_getDamperPosition();
-static int32_t gpio_getGearPosition();
-static uint32_t gpio_getSteeringWheelPosition();
-static int32_t gpio_getStrain();
+static daq_WheelSpeed_t gpio_getWheelSpeed();
```
Comment (CameronLR) [inline Teensy/src/gpio/gpio.cpp:14]: Needs it's own datatype to store all 4 wheel speeds

File: Teensy/src/gpio/gpio.cpp [hunk 3]
```diff
@@ -11,59 +11,59 @@
...
+static daq_EngineRev_t gpio_getEngineRevs();
+static daq_DamperPos_t gpio_getDamperPosition();
+static daq_GearPos_t gpio_getGearPosition();
+static daq_SteeringWhlPos_t gpio_getSteeringWheelPosition();
+static daq_Strain_t gpio_getStrain();
```
Comment (CameronLR) [inline Teensy/src/gpio/gpio.cpp:19]: Needs it's own data type to store 20 strain measurements, however, not essential right now as I doubt we'll be able to gather any strain data this year. We'll probably need to make a seperate stripped down version of the DAQ system just to gather strain for one test day down the line.

File: Teensy/src/gpio/gpio.cpp [hunk 4]
```diff
@@ -11,59 +11,59 @@
...
+static daq_FuelPressure_t gpio_getFuelPressure();
 
 bool updateSensorInfo(sensorData_s *pSensorData)
 {
-    pSensorData->wheel_speed = gpio_getWheelSpeed();
-    pSensorData->engine_revs = gpio_getEngineRevs();
-    pSensorData->damper_position = gpio_getDamperPosition();
-    pSensorData->gear_position = gpio_getGearPosition();
-    pSensorData->steering_wheel_position = gpio_getSteeringWheelPosition();
+    pSensorData->wheelSpeed_mph = gpio_getWheelSpeed();
```
Comment (CameronLR) [inline Teensy/src/gpio/gpio.cpp:27]: Wondering if m/s would be a more useful unit. Undecided on this though!

File: Teensy/src/gpio/gpio.cpp [hunk 5]
```diff
@@ -11,59 +11,59 @@
...
+    pSensorData->engineRev_rpm = gpio_getEngineRevs();
+    pSensorData->damperPos_mm = gpio_getDamperPosition();
+    pSensorData->gearPos = gpio_getGearPosition();
+    pSensorData->steeringWheelPos_degrees = gpio_getSteeringWheelPosition();
     pSensorData->strain = gpio_getStrain();
     pSensorData->gyro = gpio_getGyro();
-    pSensorData->VBat = gpio_getVBat();
-    pSensorData->throttle_position = gpio_getThrottlePosition();
-    pSensorData->fuel_pressure = gpio_getFuelPressure();
+    pSensorData->batteryVoltage_dV = gpio_getVBat();
```
Comment (CameronLR) [inline Teensy/src/gpio/gpio.cpp:34]: why 'dV', not just 'V'?

**Review by CameronLR — CHANGES_REQUESTED** [review]

File: Teensy/src/gpio/gpio.cpp (hunk 5 above)
Comment (MellingJJ) [reply Teensy/src/gpio/gpio.cpp:34]: I set it to dV for deci-volts as you put that as the units in a comment in `dap_common.h`

File: Teensy/src/gpio/gpio.cpp (hunk 4 above)
Comment (MellingJJ) [reply Teensy/src/gpio/gpio.cpp:27]: I guess it doesn't matter too much, the server can just convert it to whatever unit the user wants?

File: Teensy/src/gpio/gpio.cpp (hunk 3 above)
Comment (MellingJJ) [reply Teensy/src/gpio/gpio.cpp:19]: Ok cool, I've decided just to remove it in that case :)

File: Teensy/src/gpio/gpio.cpp (hunk 1 above)
Comment (MellingJJ) [reply Teensy/src/gpio/gpio.cpp:16]: Yep, realised that just after I put it for review, haha. Changed now

File: Teensy/src/gpio/gpio.cpp (hunk 2 above)
Comment (MellingJJ) [reply Teensy/src/gpio/gpio.cpp:14]: Changed :)

File: Teensy/src/gpio/gpio.cpp (hunk 5 above)
Comment (CameronLR) [reply Teensy/src/gpio/gpio.cpp:32]: Ahh I completely forgot about that, but makes sense as we're using integers

**Review by CameronLR — APPROVED** [review]

**Pull request merged**